PlanningBiblio/PlanningBiblio
# MT 28388: Retrieve absences from microsoft graph API

Installation:

 - edit your .env.[dev.|test.|]local (see APP_ENV in .env), and add
   the following variables:
   MS_GRAPH_TENANT_ID
   MS_GRAPH_CLIENT_ID
   MS_GRAPH_CLIENT_SECRET

 - optional variables:
   MS_GRAPH_REASON_NAME=Outlook [default: Outlook]
   MS_GRAPH_LOGIN_SUFFIX=@domain.tld [default: null]

 - in the same file, edit the following variables:
   DATABASE_URL
   DATABASE_PREFIX

 - apply the database update

 - composer update

Usage:

php bin/console PlanningBiblio:ImportMSGraph [--full=1] [--stdout=1]

The script is silent by default, see the logs database table for processing informations.

 --full=1: Performs a full import
 --stdout=1: Also output logs in stdout

Test plan:

 - Modify one or several users' email addresses to match addresses that exist
   in the Active Directory.

 - Add one or several events for those users.

 - Start the script.

 - Check that these events have been added in PLB.

 - Modify one or several events, start the script, check that those events have
   been updated in PLB.

 - Delete one or several event, start the script, check that those events have
   been deleted in PLB.

 - Check that only the events the user is an organizer of or has accepted the
   organizer's invitation are added.

 - In the case of recurring events, check that the recursion pattern is correctly
   filled in PLB.
